modules/core/templates/loginuserpass.twig

```diff
@@ -38,7 +38,7 @@
     <p>{{ 'A service has requested you to authenticate yourself. Please enter your username and password in the form below.'|trans }}</p>
     <br>
     <div class="center">
-        <form class="pure-form pure-form-aligned center-form" action="?" method="post" name="f">
+        <form class="pure-form pure-form-aligned center-form" action="{{ '?AuthState' ~ AuthState }}" method="post" name="f">
```

Comment:
This misses a `=` after ?AuthState, and likely we should be urlencoding the value of AuthState?

Reply:
Good catch! I've thought about urlencoding, but we don't do that anywhere else where we put the AuthState in the url...  I wasn't sure it's really necessary..
I think the easiest way to ensure that it doesn't cause any problems is to ensure the AuthStateID is always url-safe to begin with..

Reply:
This should be url-safe, right?
https://github.com/simplesamlphp/simplesamlphp/blob/master/lib/SimpleSAML/Utils/Random.php#L25

Reply:
It should be. But I think the really safe approach is to urlencode things at the point where we're putting them in urls. E.g. for the case someone changes the algo in random, not aware that there's an implicit requirement that it's url safe. This requirement is also not guaranteed by documentation or unit tests..

Reply:
It's actually an  easy fix.. Our moduleURL filter can deal with parameters and ensures proper urlencoding.
I will add a task on my backlog to check and fix every single template